Flashcards/task/src/flashcards/Main.java

```diff
@@ -47,133 +90,248 @@ private void runGameByAction() {
+            return;
+        }
+
+        outputMsgAndLog("The definition of the card:");
+        String cardDefinition = userScanner.nextLine();
+        applicationLogger(cardDefinition);
+
+        if (findCardDefinitionInList(cardDefinition) != null) {
+            outputMsgAndLog("The definition \"" + cardDefinition + "\" already exists.");
+            return;
+        }
+
+        Flashcard flashcard = new Flashcard(cardQuestion, cardDefinition, 0);
+        flashcardList.add(flashcard);
+        outputMsgAndLog("The pair (\"" + cardQuestion + "\"" + ":" + "\"" + cardDefinition + "\") has been added.");
+    }
+
+    private Flashcard findCardQuestionInList(String cardQuestion) {
+        for (Flashcard flashcard : flashcardList) {
+            if (flashcard.getQuestion().equals(cardQuestion)) {
+                return flashcard;
+            }
+        }
+        return null;
+    }
+
+    private Flashcard findCardDefinitionInList(String cardDefinition) {
+        for (Flashcard flashcard : flashcardList) {
+            if (flashcard.getDefinition().equals(cardDefinition)) {
+                return flashcard;
+            }
+        }
+        return null;
+    }
+
+    private void resetStats() {
+        for (Flashcard flashcard : flashcardList) {
+            flashcard.resetStats();
+        }
+        outputMsgAndLog("Card statistics has been reset.");
+    }
+
+    private boolean hardestCardPresent() {
+        for (Flashcard flashcard : flashcardList) {
+            if (flashcard.getErrors() > 0) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void hardestCard() {
+        if (!hardestCardPresent()) {
+            outputMsgAndLog("There are no cards with errors.");
+            return;
+        }
+
+        ArrayList<String> hardestCardList = new ArrayList<>();
+        int hardestCardNumber = 0;
+
+        for (Flashcard flashcard : flashcardList) {
+            if (flashcard.getErrors() > hardestCardNumber) {
+                hardestCardList.clear();
+                hardestCardList.add(flashcard.getQuestion());
+                hardestCardNumber = flashcard.getErrors();
+            } else if (flashcard.getErrors() == hardestCardNumber) {
+                hardestCardList.add(flashcard.getQuestion());
+            }
+        }
+        outputMsgAndLog(createHardestCardOutput(hardestCardList, hardestCardNumber));
+    }
+
+    private String createHardestCardOutput(ArrayList<String> hardestCardList, int hardestCardNumber) {
+        if (hardestCardList.size() == 1) {
+            return ("The hardest card is \"" + hardestCardList.get(0) + "\". You have "
+                    + hardestCardNumber + " errors answering it.");
+        }
+
+        String hardestCardOutput = hardestCardList.get(0);
+        for (int i = 1; i < hardestCardList.size(); i++) {
+            hardestCardOutput = hardestCardOutput.concat("\", \"" + hardestCardList.get(i));
```

Comment:
Smaakkwestie. Mijn voorkeur gaat naar de for-loop van i = 0 naar size() te laten lopen en daarna de laatste ", " weg te halen.

Reply:
Done.

Reply:
Done.